### ISSUE TITLE
Throwing Errors adding line series elements to canvas

### ISSUE THREAD

**Issue by xanth** [beto-rodriguez/Live-Charts#70]
Thanks to @beto-rodriguez [my code](https://github.com/beto-rodriguez/Live-Charts/issues/59#issuecomment-170770905) hasn't really changed since the last issue I came across. Sadly I've hit a new error where the UI crashes when a view with a Live-Chart on it is navigated to twice.
To reproduce this I do the following:
1. Load WPF application -> Home Page is displayed (Live Chart)
2. Navigate to other view (no live chart)
3. Navigate back to home page
4. Navigate to other view (no live chart)
5. Navigate back to home page -> Crash

This is the stack trace I'm getting:
```
at System.Windows.FrameworkElement.AddLogicalChild(Object child)
at System.Windows.FrameworkElement.ChangeLogicalParent(DependencyObject newParent)
at System.Windows.Controls.UIElementCollection.AddInternal(UIElement element)
at System.Windows.Controls.UIElementCollection.Add(UIElement element)
at LiveCharts.Charts.Chart.PrepareCanvas(Boolean animate) in C:\Dev\Project\Trunk\Sources\Charts\Charts\Chart.cs:line 739
at LiveCharts.Charts.Chart.ClearAndPlot(Boolean animate) in C:\Dev\Project\Trunk\Sources\Charts\Charts\Chart.cs:line 249
at LiveCharts.Charts.Chart.InitializeSeries(Chart chart) in C:\Dev\Project\Trunk\Sources\Charts\Charts\Chart.cs:line 830
at LiveCharts.Charts.Chart.SeriesChangedCallback(DependencyObject o, DependencyPropertyChangedEventArgs eventArgs) in C:\Dev\Compliance Check\Trunk\Sources\Charts\Charts\Chart.cs:line 804
at System.Windows.DependencyObject.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
at System.Windows.FrameworkElement.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
at System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args)
at System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)
at System.Windows.FrameworkTemplate.ReceivePropertySet(Object targetObject, XamlMember member, Object value, DependencyObject templatedParent)
at System.Windows.FrameworkTemplate.<>c__DisplayClass0.<LoadOptimizedTemplateContent>b__6(Object sender, XamlSetValueEventArgs setArgs)
at System.Xaml.XamlObjectWriter.OnSetValue(Object eventSender, XamlMember member, Object value)
at System.Xaml.XamlObjectWriter.Logic_ApplyPropertyValue(ObjectWriterContext ctx, XamlMember prop, Object value, Boolean onParent)
at System.Xaml.XamlObjectWriter.Logic_DoAssignmentToParentProperty(ObjectWriterContext ctx)
at System.Xaml.XamlObjectWriter.Logic_AssignProvidedValue(ObjectWriterContext ctx)
at System.Xaml.XamlObjectWriter.WriteEndObject()
at System.Xaml.XamlWriter.WriteNode(XamlReader reader)
at System.Windows.FrameworkTemplate.LoadTemplateXaml(XamlReader templateReader, XamlObjectWriter currentWriter)
```

I've found that if I comment out lines 737 to 743 in Chart.cs (version 0.5.10) the error doesn't show itself however I suspect this will inhibit me from displaying additional Series or changes to the collections.

**Comment by beto-rodriguez** [comment]
Thank you very much for this detailed report, 0.6 will be released soon, it is a full review of live charts I have made important changes and I am really happy with the result, it should be much more stable and scalable, I will fix this as soon as 0.6 is ready.  I am sure you will love 0.6 much easier and powerful sintax

**Comment by beto-rodriguez** [comment]
I will try to force this error, just one question when you say "load view" you mean a new window?, or just a partial change in current window? Maybe using a ContentControl

**Comment by xanth** [comment]
partial change in current window, we have a shell window where we use a content control bound to a viewmodel to switch the views with Caliburn.Micro.

**Comment by beto-rodriguez** [comment]
0.6 is a full review of the code, it is now much cleaner I have not been able to force this bug, are you still getting it with last version, 0.6.1 is even cleaner, I would appreciate if you test it with it (not nugget yet)

**Comment by xanth** [comment]
I've updated the project to use 0.6.1 and the bug still showed itself.

```
   at System.Windows.FrameworkElement.ChangeLogicalParent(DependencyObject newParent)
   at System.Windows.FrameworkElement.AddLogicalChild(Object child)
   at System.Windows.Controls.UIElementCollection.AddInternal(UIElement element)
   at System.Windows.Controls.UIElementCollection.Add(UIElement element)
   at lvc.Charts.Chart.PrepareCanvas(Boolean animate) in C:\Dev\Project\Trunk\Sources\Charts\CoreComponents\Chart.cs:line 735
   at lvc.Charts.Chart.ClearAndPlot(Boolean animate) in C:\Dev\Project\Trunk\Sources\Charts\CoreComponents\Chart.cs:line 265
   at lvc.Charts.Chart.InitializeSeries(Chart chart) in C:\Dev\Project\Trunk\Sources\Charts\CoreComponents\Chart.cs:line 846
   at lvc.Charts.Chart.SeriesChangedCallback(DependencyObject o, DependencyPropertyChangedEventArgs eventArgs) in C:\Dev\Project\Trunk\Sources\Charts\CoreComponents\Chart.cs:line 800
   at System.Windows.DependencyObject.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
   at System.Windows.FrameworkElement.OnPropertyChanged(DependencyPropertyChangedEventArgs e)
   at System.Windows.DependencyObject.NotifyPropertyChange(DependencyPropertyChangedEventArgs args)
   at System.Windows.DependencyObject.UpdateEffectiveValue(EntryIndex entryIndex, DependencyProperty dp, PropertyMetadata metadata, EffectiveValueEntry oldEntry, EffectiveValueEntry& newEntry, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType)
   at System.Windows.FrameworkTemplate.ReceivePropertySet(Object targetObject, XamlMember member, Object value, DependencyObject templatedParent)
   at System.Windows.FrameworkTemplate.<>c__DisplayClass0.<LoadOptimizedTemplateContent>b__6(Object sender, XamlSetValueEventArgs setArgs)
   at System.Xaml.XamlObjectWriter.OnSetValue(Object eventSender, XamlMember member, Object value)
   at System.Xaml.XamlObjectWriter.Logic_ApplyPropertyValue(ObjectWriterContext ctx, XamlMember prop, Object value, Boolean onParent)
   at System.Xaml.XamlObjectWriter.Logic_DoAssignmentToParentProperty(ObjectWriterContext ctx)
   at System.Xaml.XamlObjectWriter.Logic_AssignProvidedValue(ObjectWriterContext ctx)
   at System.Xaml.XamlObjectWriter.WriteEndObject()
   at System.Xaml.XamlWriter.WriteNode(XamlReader reader)
   at System.Windows.FrameworkTemplate.LoadTemplateXaml(XamlReader templateReader, XamlObjectWriter currentWriter)
```

Then we modified the containing view model to inherit from ViewAware rather than just PropertyChangedBase and that fixed our issue.

**Comment by beto-rodriguez** [comment]
Thanks for reporting, I will work on it

**Comment by beto-rodriguez** [comment]
I have not test this but I think I founf the solution

**Comment by beto-rodriguez** [comment]
specially this lines should solve the problem

https://github.com/beto-rodriguez/Live-Charts/commit/cb819deded4da08d3999280780fa667ad2fb903b#diff-7cabf91fc05cf7e480b95dffd3ac5605R883

https://github.com/beto-rodriguez/Live-Charts/commit/cb819deded4da08d3999280780fa667ad2fb903b#diff-c148adab5d542cc147f6f9a3ccc9e63cR242

**Comment by thomasjoscht** [comment]
Hey there, I'm getting nearly same error by navigating second time to my view. Here is my exception:

System.InvalidOperationException: Bei dem angegebenen Element handelt es sich bereits um das logische untergeordnete Element eines anderen Elements. Führen Sie zuerst eine Trennung durch.
   bei System.Windows.FrameworkElement.ChangeLogicalParent(DependencyObject newParent)
   bei System.Windows.FrameworkElement.AddLogicalChild(Object child)
   bei System.Windows.Controls.UIElementCollection.AddInternal(UIElement element)
   bei LiveCharts.CoreComponents.Chart.PrepareCanvas(Boolean animate)
   bei LiveCharts.CoreComponents.Chart.InitializeSeries(Chart chart)
   bei LiveCharts.CoreComponents.Chart.PrepareCanvas(Boolean animate)
   bei System.Windows.Threading.DispatcherTimer.FireTick(Object unused)
   bei System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   bei System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

Have you fixed this anyway? Can I help you investigating this error? Thx.

**Comment by beto-rodriguez** [comment]
I  am practically sure th will dissapear in 0.6.4 I hope I can publish 0.6.4 today

**Comment by thomasjoscht** [comment]
That will be great. Thanks a lot!

**Comment by beto-rodriguez** [comment]
I already tested this with new UnitTesting project. this is solved. here is a test mocking this case, https://github.com/beto-rodriguez/Live-Charts/blob/master/UnitTesting/General/SeriesSharedInCharts.cs

the problem was that old canvas was staying in memory and had your series as VisualChilds (I am not really sure how), but now as you can see in the test I force this case and the series should only stay in the second chart. I have not tested this in UI but I am sure it will only plot in second chart wich is what normally wpf does in this cases.

Thanks again for reporting, let me know if you still have any troubles with this issue.

P.S. this will be published with 0.6.4

**Comment by thomasjoscht** [comment]
Thank you! I've tested it in my application and it works correctly!
